rust/src/error.rs

```diff
@@ -45,6 +45,8 @@ pub enum Error {
     },
     #[snafu(display("Not supported: {source}"))]
     NotSupported { source: BoxedError },
+    #[snafu(display("Invalid Operation {message}"))]
+    InvalidOperation { message: String },
```

Comment:
I think we can avoid needing this new error variant for now by just coercing the `WriteMode` when append is called.

```suggestion
```